File: backend/FitnessApp/src/main/java/com/andrea360/fitnessapp/model/Purchase.java
package com.andrea360.fitnessapp.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@NoArgsConstructor
@Getter
@Setter
@Entity
public class Purchase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false, unique = true)
    private String stripePaymentIntentId;

    @ManyToOne(optional = false)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @ManyToOne(optional = false)
    @JoinColumn(name = "service_id", nullable = false)
    private TrainingType trainingType;

    @Positive
    @Column(nullable = false)
    private int quantity;

    @PositiveOrZero
    @Column(nullable = false)
    private int remaining;

    @Column(nullable = false)
    private LocalDateTime purchasedAt;

}

File: frontend/src/types/Purchase.ts
export interface Purchase {
  id: number;
  trainingTypeName: string;
  trainingTypePrice: number;
  totalPrice: number;
  quantity: number;
  remaining: number;
  purchasedAt: string;
}

export interface CreatePurchaseRequest {
  memberId: number;
  trainingTypeId: number;
  quantity: number;
  stripePaymentIntentId: string;
}